client/src/modules/journal/modals/search.modal.js

```diff
@@ -22,11 +27,38 @@ function JournalSearchModalController(Instance, Projects, Notify, Store, filters
   // assign already defined custom filters to searchQueries object
   vm.searchQueries = util.maskObjectFromKeys(filters, searchQueryOptions);
 
+  /**
+   * hasDefaultAccount is used to set a default account selection behavior
+   * if the search modal need to set account selection in default query panel we can send it
+   * as parameters
+   * @example
+   * <pre>
+   * Config.openSearchModal(filters, { hasDefaultAccount : true })
+   * </pre>
+   */
+  if (options.hasDefaultAccount) {
+    vm.hasDefaultAccount = true;
+  }
+
   // assign default filters
   if (filters.limit) {
     vm.defaultQueries.limit = filters.limit;
   }
 
+  // assing default account
+  if (filters.account_id) {
+    vm.defaultQueries.account_id = filters.account_id;
+  }
+
+  Account.read()
```

Comment:
You don't need this .. I don't think.  The `onSelectAccount()` callback should return the entire account object.  Just do this:

```js
vm.onSelectAccountCallback = function (account) {
  vm.currentAccountLabel = account.hrLabel;
  // other logic
});

angular.forEach(vm.searchQueries, function (value, key) {
  if (angular.isDefined(value)) {
    if (key === 'account_id') {
      changes.post({ key : key, value : value, displayValue : vm.currentAccountLabel });
    }
    // ... etc ...
  }
});
```

This way, you don't have to download the entire account list.